ocaml/xapi/xapi_vif.ml

```diff
@@ -55,13 +56,31 @@ let refresh_filtering_rules ~__context ~self =
 
 (* This function moves a dom0 vif device from one bridge to another, without involving the guest,
  * so it also works on guests that do not support hot(un)plug of VIFs. *)
-let move ~__context ~network vif =
+let move_internal ~__context ~network ?active vif =
 	debug "Moving VIF %s to network %s" (Db.VIF.get_uuid ~__context ~self:vif)
 		(Db.Network.get_uuid ~__context ~self:network);
+	let active =
+		match active with
+		| None -> device_active ~__context ~self:vif
+		| Some x -> x
+	in
 	Db.VIF.set_network ~__context ~self:vif ~value:network;
-	if device_active ~__context ~self:vif
+	if active
 	then Xapi_xenops.vif_move ~__context ~self:vif network
 
+let move ~__context ~self ~network =
+	let active = device_active ~__context ~self in
+	let vm = Db.VIF.get_VM ~__context ~self in
+	let host = Db.VM.get_resident_on ~__context ~self:vm in
```

Comment:
You only need `vm` and `host` in case `active = true`, so we can save two DB calls if it is not:
```
if active then begin
    let vm = ...
    let host = ...
    ...
end;
```